tests/cases/Mapping/RequestParameterMapping.phpt

```diff
@@ -0,0 +1,290 @@
+<?php
+
+/**
+ * Test: Mapping\RequestParameterMapping
+ */
+
+require_once __DIR__ . '/../../bootstrap.php';
+
+use Apitte\Core\Http\ApiRequest;
+use Apitte\Core\Http\ApiResponse;
+use Apitte\Core\Http\RequestAttributes;
+use Apitte\Core\Mapping\Parameter\FloatTypeMapper;
+use Apitte\Core\Mapping\Parameter\IntegerTypeMapper;
+use Apitte\Core\Mapping\Parameter\StringTypeMapper;
+use Apitte\Core\Mapping\RequestParameterMapping;
+use Apitte\Core\Schema\Endpoint;
+use Apitte\Core\Schema\EndpointParameter;
+use GuzzleHttp\Psr7\Response;
+use GuzzleHttp\Psr7\ServerRequest;
+use Tester\Assert;
+use Tester\TestCase;
+
+final class TestRequestParameterMapping extends TestCase
+{
+
+	/**
+	 * @var RequestParameterMapping
+	 */
+	private $requestParameterMapping;
+
+	/**
+	 * @var ApiRequest
+	 */
+	private $request;
+
+	/**
+	 * @var ApiResponse
+	 */
+	private $response;
+
+	protected function setUp()
+	{
+		$this->requestParameterMapping = new RequestParameterMapping;
+
+		$this->requestParameterMapping->addMapper('string', new StringTypeMapper);
+		$this->requestParameterMapping->addMapper('int', new IntegerTypeMapper);
+		$this->requestParameterMapping->addMapper('float', new FloatTypeMapper);
+
+		$this->request = new ApiRequest(
+			new ServerRequest(
+				'GET',
+				'/'
+			)
+		);
+		$this->response = new ApiResponse(new Response);
+	}
+
+	public function testIntInPath()
+	{
+		$endpoint = new Endpoint;
+
+		$idEndpointParameter = new EndpointParameter;
+		$idEndpointParameter->setName('id');
+		$idEndpointParameter->setType('int');
+		$idEndpointParameter->setIn($idEndpointParameter::IN_PATH);
+		$idEndpointParameter->setRequired(FALSE);
+		$idEndpointParameter->setAllowEmpty(TRUE);
+
+		$endpoint->addParameter($idEndpointParameter);
+
+		$request = $this->request
+			->withAttribute(RequestAttributes::ATTR_ENDPOINT, $endpoint)
+			->withAttribute(RequestAttributes::ATTR_PARAMETERS, [
+				'id' => NULL,
+			]);
+
+		// ---- test optional parameter
+
+		$notRequiredIdResponse = $this->requestParameterMapping->map(
+			$request,
+            $this->response
+		);
+
+		Assert::null($notRequiredIdResponse->getAttribute(RequestAttributes::ATTR_PARAMETERS)['id']);
+
+		// ---- test throw missing parameter
+
+		$idEndpointParameter->setRequired(TRUE);
+
+		Assert::throws(
+            function () use ($request) {
+                $this->requestParameterMapping->map($request, $this->response);
+            },
+            \Apitte\Core\Exception\Logical\InvalidStateException::class,
```

Comment:
Could you please import these exceptions? Thanks.

Reply:
Sure, it is done